File: backend/src/TalentFlow.Domain/Entities/Job.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;
using static System.Net.Mime.MediaTypeNames;

namespace TalentFlow.Domain.Entities;

public class Job
{
    public Guid Id { get; set; }

    public Guid CompanyId { get; set; }

    public Guid DepartmentId { get; set; }

    [Required]
    [MaxLength(255)]
    public string Title { get; set; } = null!;

    public string? Description { get; set; }

    public string? Responsibilities { get; set; }

    public string? Requirements { get; set; }

    [MaxLength(50)]
    public string? EmploymentType { get; set; }

    [MaxLength(50)]
    public string? ExperienceLevel { get; set; }

    public decimal? SalaryMin { get; set; }

    public decimal? SalaryMax { get; set; }

    [MaxLength(150)]
    public string? Location { get; set; }

    public bool IsRemote { get; set; }

    public DateTime? ApplicationDeadline { get; set; }

    public bool IsActive { get; set; }

    [MaxLength(50)]
    public string? Status { get; set; }

    public bool IsDeleted { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    // Navigation Properties
    public Company Company { get; set; } = null!;
    public Department Department { get; set; } = null!;
    public ICollection<Application> Applications { get; set; } = new List<Application>();
}

File: frontend/src/types/job.ts


export interface Company {
    id: string;
    name: string;
    description?: string;
    industry?: string;
    city?: string;
    country?: string;
}

export interface Department {
    id: string;
    companyId: string;
    name: string;
    description?: string;
}

export interface Job {
    id: string;
    companyId: string;
    departmentId: string;
    title: string;
    description?: string;
    responsibilities?: string;
    requirements?: string;
    employmentType?: string;
    experienceLevel?: string;
    salaryMin?: number;
    salaryMax?: number;
    location?: string;
    isRemote: boolean;
    applicationDeadline?: string;
    isActive: boolean;
    status?: string;
    companyName: string;
    departmentName: string;
    applicationCount: number;
    createdAt: string;
    updatedAt: string;
}

export interface CreateJobRequest {
    companyName: string;      
    departmentName: string;
    title: string;
    description?: string;
    responsibilities?: string;
    requirements?: string;
    employmentType?: string;
    experienceLevel?: string;
    salaryMin?: number;
    salaryMax?: number;
    location?: string;
    isRemote: boolean;
    applicationDeadline?: string;
}

export interface UpdateJobRequest {
    companyId: string;
    departmentId: string;
    title: string;
    description?: string;
    responsibilities?: string;
    requirements?: string;
    employmentType?: string;
    experienceLevel?: string;
    salaryMin?: number;
    salaryMax?: number;
    location?: string;
    isRemote: boolean;
    applicationDeadline?: string;
    isActive: boolean;
    status?: string;
}

export interface Application {
    id: string;
    candidateId: string;
    jobId: string;
    status?: string;
    appliedAt: string;
    coverLetter?: string;
    candidateName: string;
    jobTitle: string;
    companyName: string;
    createdAt: string;
    updatedAt: string;
}

export interface CreateApplicationRequest {
    jobId: string;
    coverLetter?: string;
}

export interface UpdateApplicationStatusRequest {
    status: string;
}